src/main/java/bio/terra/app/controller/SnapshotsApiController.java

```diff
@@ -200,6 +201,17 @@ private AuthenticatedUserRequest getAuthenticatedInfo() {
     return new ResponseEntity<>(snapshotModel, HttpStatus.OK);
   }
 
+  @Override
+  public ResponseEntity<SnapshotPreviewModel> lookupSnapshotPreviewById(
+      UUID id, String table, Integer count) {
+    logger.info("Verifying user access");
+    iamService.verifyAuthorization(
+        getAuthenticatedInfo(), IamResourceType.DATASNAPSHOT, id.toString(), IamAction.READ_DATA);
+    logger.info("Retrieving snapshot");
+    SnapshotPreviewModel previewModel = snapshotService.retrievePreview(id, table, count);
```

Comment:
Since `count` is user provided maybe a max value is warranted here?
```suggestion
    SnapshotPreviewModel previewModel = snapshotService.retrievePreview(id, table, Math.min(count, MAX_ROW_COUNT));
```

Reply:
Oh yeah, this is a big one.